File: 2415-reverse-odd-levels-of-binary-tree/2415-reverse-odd-levels-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode reverseOddLevels(TreeNode root) {
        
      Deque<TreeNode> queue=new LinkedList();
        int level = 0;
        int tmp;

        queue.add(root);
        while(!queue.isEmpty()) {
            int size = queue.size();
            if(level%2 == 1) {  // reversing the current level vals
                Deque<TreeNode> queue1 = new LinkedList(queue);
                while(!queue.isEmpty())
                {
                    tmp = queue.getFirst().val;
                    queue.getFirst().val = queue.getLast().val;
                    queue.getLast().val = tmp;
                    queue.pollFirst();
                    queue.pollLast();
                }
                queue=queue1;
            }

            if(queue.peek().left == null)     break;

            for(int i=0;i<size;i++) {
                TreeNode node = queue.poll();
                queue.add(node.left);
                queue.add(node.right);
            }
            level++;
        }
        return root;
    }
}

File: 2415-reverse-odd-levels-of-binary-tree/2415-reverse-odd-levels-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    val=0
    def reverseOddLevels(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        def n(n1,n2, level):
            if (not n1) or (not n2): return
            if (level % 2) == 1: 
                n1.val, n2.val = n2.val, n1.val
            n(n1.left,n2.right, level + 1)
            n(n1.right,n2.left, level + 1)

        n(root.left,root.right, 1)
        return root